File: Kitchen Routing System/UI/KitchenUI/src/app/Models/Order.ts
import { Customer } from "./Customer";
import { IOrderItem } from "./IOrderItem";
import { OrderStatus } from "./OrderStatus.enum";

export class Order {
  id: number;
  items: IOrderItem[];
  status: OrderStatus;
  customerDetails: Customer;
  createdDate: Date;

  constructor(id: number, items: IOrderItem[], customerName: string, customerEmail: string, customerPhone: string) {
    this.id = id;
    this.items = items;
    this.status = OrderStatus.Pending;
    this.customerDetails = new Customer(customerName, customerEmail, customerPhone);
    this.createdDate = new Date();
  }
}

File: Kitchen Routing System/Backend/KitchenServerAPI/Models/Order.cs
namespace KitchenServerAPI.Models
{
    public class Order
    {
        int Id { get; set; }
        public IOrderItem[] OrderItems { get; set; }
        OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
        DateTime CreatedDate { get; set; }
        public Customer CustomerDetails { get; set; }

        public Order(int id, OrderStatus orderStatus)
        {
            this.Id = id;
            this.OrderStatus = orderStatus;
            this.CreatedDate = DateTime.Now;
        }
    }
}
